### ISSUE TITLE
Not getting access token

### ISSUE THREAD

**Issue by amarnathv19** [sahat/satellizer#1027]
Hi,

Thanks for the great plugin. I am using facebook and google login using this, 

$auth.authenticate('google') returns response in two ways

{acess_token: : xxx, expires_in: "3600"}

But sometimes I am not getting access_token instead I am getting {login:true"}

Can you guide me how to fix this?, I am using this plugin from cdn

Thanks

**Comment by zmaximo1990** [comment]
Hi, 

I have the same issue. Sometimes, without conditions, $auth.authenticate('google') or $auth.authenticate('facebook') with responseType='token' got an empty object after autenticate.

Please i need help!, i need to solve this problem!. I am going to deploy an application in a few days!.
We are using versión 0.15.5

any idea?.

Thanks a lot.

**Comment by ghost** [comment]
I'm also having this issue.

```
        $authProvider.google({
            clientId: 'xxx.apps.googleusercontent.com',
            responseType: 'token',
            scope: ['openid', 'profile', 'email']
        });
```

```
        $scope.authenticate = function(provider) {
            $auth.authenticate(provider)
                .then(function(response){
                    console.log(response);
                })
        };
```

response from console.log() is just {login: true}.

@zmaximo1990 @amarnathv19 Did you guys figure it out?

**Comment by zmaximo1990** [comment]
@kosiraljaz no. We just used the authorization code grant type and exchange the access_token in server side to get user details. 
You are getting "{login: true}" but we receive only "{}".
Maybe you can use the access_token saved in $localStorage by Satellizer using $auth.getToken().

If you can solve it, write in this post please. Thanks.

Good look.

**Comment by ghost** [comment]
@zmaximo1990 
I solved this issue by declaring new state:

```
            .state('root', {
                url: '/state={state}&access_token={access_token}',
                template: '',
            })
```
this is the only code you'll need to make it work, everything else is handled by satellizer. You can make more complex regex verifications for query params if you want.

**Comment by condereis** [comment]
I am having the same issue. @kosiraljaz solution didn't help me. Any news?

**Comment by mini2728** [comment]
I solved this issue

I have used $urlRouterProvider.otherwise('/login');
but my Router not setting 
need setting
$stateProvider
            .state('app.app-login-page', {
              url: '/state={state}*path',
              views: {
                  'content@': {
                      templateUrl: '<div></div>'
                  }
              }
            })
